### PR TITLE
usbhost: test for ehci and mass storage crates

### PR DESCRIPTION
**Pull request by badicsalex**
Related repositories:

https://github.com/Foundation-Devices/ehci/
https://github.com/Foundation-Devices/mass-storage/

I'm not sure how to handle the review of those. The interface for `ehci` will most probably change during integration into %projectname%

### CODE REVIEW

File: src/logging.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    fn flush(&self) {}
+}
+
+
+
+// Dummy critical section code that doesn't really do anything.
+// If we screw up with an interrupt handler, well, oops.
+struct MyCriticalSection;
```
Comment (eupn) [inline src/logging.rs:51]: We have a critical section [implemented](https://github.com/Foundation-Devices/armv7/blob/update/src/critical_section.rs) in `armv7` crate. It's behind the `critical-section` feature. You can try importing that one instead.

**Comment by eupn** [comment]
@badicsalex LGTM so far, is there anything I'd need to try to run it on my board? I have a thumb drive I can try with it.

File: src/bin/usbhost.rs [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+    log::{debug, error, info, trace, warn},
+};
+
+global_asm!(include_str!("../start.S"));
+
+// MCK: 164MHz
+// Clock frequency is divided by 2 because of the default `h32mxdiv` PMC setting
+const MASTER_CLOCK_SPEED: u32 = 164000000 / 2;
+
+const UHPHS_BASE: usize = 0x00500000;
```
Comment (eupn) [inline src/bin/usbhost.rs:37]: This base address should come from the SVDs and generated `utralib` code but it's not there. I've asked @jeandudey to update the SVDs to have the base address included. Before we have it, it's okay to have this address hardcoded.

**Comment by badicsalex** [comment]
@eupn 

It should work with any pendrive. I tried it out with only 2, with a USB-C -> USB-A(female) adapter.

**Review by eupn — APPROVED** [review]

**Pull request merged**